nixos/default.nix

```
@@ -178,7 +181,8 @@
       toURL = mkOption {
         description = "URL to forward HTTP requests to. Either this or the urlParts option must be set.";
         type = types.str;
-        default = "${config.urlParts.protocol}://${config.urlParts.host}:${builtins.toString config.urlParts.port}";
+        default = "${config.urlParts.protocol}://${config.urlParts.host}:${lib.toString config.urlParts.port}";
+        defaultText = lib.literalExpression "\${config.services.tsnsrv.<name>.urlParts.protocol}://\${config.services.tsnsrv.<name>.urlParts.host}:\${lib.toString config.services.tsnsrv.<name>.urlParts.port}";
```

`nix flake check` doesn't like `lib.toString`, think it should remain builtins:

```suggestion
        default = "${config.urlParts.protocol}://${config.urlParts.host}:${builtins.toString config.urlParts.port}";
        defaultText = lib.literalExpression "\${config.services.tsnsrv.<name>.urlParts.protocol}://\${config.services.tsnsrv.<name>.urlParts.host}:\${builtins.toString config.services.tsnsrv.<name>.urlParts.port}";
```

---

```suggestion
        default = "${config.urlParts.protocol}://${config.urlParts.host}:${toString config.urlParts.port}";
        defaultText = lib.literalExpression "\${config.services.tsnsrv.<name>.urlParts.protocol}://\${config.services.tsnsrv.<name>.urlParts.host}:\${toString config.services.tsnsrv.<name>.urlParts.port}";
```

Omitting definitely also works to keep the line a bit shorter